### PR TITLE
Refresh bindings & rename x86 module

### PR DESCRIPTION
**Pull request by rbradford**
### Summary of the PR

- docs: Updating CONTRIBUTING.md
- arm64, x64: Update bindings for the existing architectures
- x86_64: Rename from x86 for consistency
- misc: Update CHANGELOG for API changes

### Requirements

Before submitting your PR, please make sure you addressed the following
requirements:

- [x] All commits in this PR are signed (with `git commit -s`), and the commit
  message has max 60 characters for the summary and max 75 characters for each
  description line.
- [x] All added/changed functionality has a corresponding unit/integration
  test.
- [x] All added/changed public-facing functionality has entries in the "Upcoming 
  Release" section of CHANGELOG.md (if no such section exists, please create one).
- [x] Any newly added `unsafe` code is properly documented.

### CODE REVIEW

**Comment by rbradford** [comment]
In #78 there was a comment about `#[derive(..)]` changes. Some of those have been replaced by automatic generation of blocks. However unfortunately I think it is very hard to maintain API stability for this crate because sometimes the kernel headers change in such a way (like the `KVM_SYSTEM_EVENT` exit data) that changes the generated API.

With a fix to kvm-ioctls to reflect the above change I was able to run the kvm-ioctls test.

**Comment by rbradford** [comment]
In #78 there was a comment about `#[derive(..)]` changes. Some of those have been replaced by automatic generation of blocks. However unfortunately I think it is very hard to maintain API stability for this crate because sometimes the kernel headers change in such a way (like the `KVM_SYSTEM_EVENT` exit data) that changes the generated API.

With a fix to kvm-ioctls to reflect the above change I was able to run the kvm-ioctls test.

**Comment by andreeaflorescu** [comment]
> With a fix to kvm-ioctls to reflect the above change I was able to run the kvm-ioctls test.

Do you have a branch with the change needed in kvm-ioctls so we can take a look?

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,4 +1,9 @@
 # Changelog
+## [Unreleased]
+
+- API change in the bindings from upstream kernel changes
```
Comment (andreeaflorescu) [inline CHANGELOG.md:4]: I looked through the changes but I can't really spot the API change, do you mind expanding a bit on what that's about?

File: src/arm64/bindings.rs [hunk 2]
```diff
@@ -4881,6 +5511,11 @@ impl Default for kvm_run {
         }
     }
 }
+impl ::std::fmt::Debug for kvm_run {
```
[the remark this answers is not in the document]
Comment (rbradford) [reply src/arm64/bindings.rs:5514]: Nope, this is generated by bindgen.

File: src/lib.rs [hunk 3]
```diff
@@ -12,9 +12,9 @@
 extern crate vmm_sys_util;
 
 #[cfg(any(target_arch = "x86", target_arch = "x86_64"))]
```
[the remark this answers is not in the document]
Comment (rbradford) [reply src/lib.rs:14]: Sure, let me update that.

**Comment by rbradford** [comment]
> > With a fix to kvm-ioctls to reflect the above change I was able to run the kvm-ioctls test.
> 
> Do you have a branch with the change needed in kvm-ioctls so we can take a look?

https://github.com/rust-vmm/kvm-ioctls/pull/218

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -17,48 +17,50 @@ repository on your machine:
...
 ```
 
-## Add a new architecture
+## Updating bindings / adding a new architecture
+
 When adding a new architecture, the bindings must be generated for all existing
 versions for consistency reasons.
 
-### Example for arm64 and version 5.13
+### Example for arm64 and version 6.2
```
[the remark this answers is not in the document]
Comment (rbradford) [reply CONTRIBUTING.md]: Done

File: CHANGELOG.md (hunk 1 above)
Comment (rbradford) [reply CHANGELOG.md]: `system_event` became a union itself.

**Review by andreeaflorescu — DISMISSED** [review]

**Review by sboeuf — APPROVED** [review]

**Comment by rbradford** [comment]
Rebased. @sboeuf and @andreeaflorescu can I get a re-approval?

File: CHANGELOG.md (hunk 1 above)
Comment (andreeaflorescu) [reply CHANGELOG.md:4]: I think it would be better to mention this explicitly in the changelog as it was in the previous revision.

Comment (rbradford) [reply CHANGELOG.md]: I had multiple checkouts and used the wrong version. This should be correct now.

File: CHANGELOG.md [hunk 5]
```diff
@@ -1,4 +1,9 @@
...
+- The x86 module has been renamed to x86_64 for consistency (matches the kernel
```
[the remark this answers is not in the document]
Comment (rbradford) [reply CHANGELOG.md:6]: Sorry, I did include it in the version you reviewed but I rebased the wrong version.

File: README.md [hunk 6]
```diff
@@ -5,7 +5,6 @@
 Rust FFI bindings to KVM, generated using
 [bindgen](https://crates.io/crates/bindgen). It currently has support for the
 following target architectures:
-- x86
```
Comment (rbradford) [inline README.md:8]: @andreeaflorescu Here it is.

**Review by andreeaflorescu — APPROVED** [review]

**Pull request merged**